toolz/functoolz.py

```diff
@@ -200,6 +200,7 @@ def __init__(self, *args, **kwargs):
 
         self.__doc__ = getattr(func, '__doc__', None)
         self.__name__ = getattr(func, '__name__', '<curry>')
+        self.__module__ = getattr(func, '__module__', None)
```

Comment:
Any reason not use `functools.wraps` or `functools.update_wrapper`? See https://docs.python.org/2.7/library/functools.html#functools.update_wrapper